src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -271,6 +273,82 @@ build_mhd_response(const struct sol_http_response *response, time_t last_modifie
     return NULL;
 }
 
+static struct MHD_Response *
+build_mhd_response(const struct sol_http_response *response, time_t last_modified)
+{
+    struct MHD_Response *r;
+
+    r = MHD_create_response_from_buffer(response->content.used, response->content.data,
+        MHD_RESPMEM_MUST_COPY);
+    if (!r)
+        return NULL;
+
+    r = build_mhd_response_params(r, response);
+    if (!r)
+        return NULL;
+
+    if (!set_last_modified_header(r, last_modified)) {
+        MHD_destroy_response(r);
+        return NULL;
+    }
+
+    return r;
+}
+
+static void
+response_progressive_del_cb(void *data)
+{
+    struct sol_http_response_progressive *progressive = data;
+
+    if (progressive->on_del)
+        progressive->on_del((void *)progressive->cb_data, progressive);
+
+    sol_buffer_fini(&progressive->buffer);
+    free(progressive);
+}
+
+static ssize_t
+response_progressive_cb(void *data, uint64_t pos, char *buf, size_t size)
+{
+    int r;
+    ssize_t len;
+    struct sol_http_response_progressive *progressive = data;
+
+    if (progressive->delete_me) {
+        if (!progressive->graceful_del ||
+            (progressive->graceful_del && !progressive->buffer.used))
+            return MHD_CONTENT_READER_END_OF_STREAM;
+    }
+
+    if (!progressive->buffer.used) {
+        MHD_suspend_connection(progressive->connection);
+        progressive->suspended = true;
+        return 0;
+    }
+
+    len = sol_min(size, progressive->buffer.used);
+    memcpy(buf, progressive->buffer.data, len);
+
+    r = sol_buffer_remove_data(&progressive->buffer, 0, len);
+    SOL_INT_CHECK(r, < 0, MHD_CONTENT_READER_END_WITH_ERROR);
+
+    return len;
+}
+
+static struct MHD_Response *
+build_mhd_response_progressive(const struct sol_http_response *response,
+    struct sol_http_response_progressive *progressive)
+{
+    struct MHD_Response *r;
+
+    r = MHD_create_response_from_callback(MHD_SIZE_UNKNOWN, 128,
```

Comment:
I'm wondering if 128 is a good number. Maybe go with `LINE_MAX` or page size (4096)?